### ISSUE TITLE
Enable use of MQTT broker as a library

### ISSUE THREAD

**Issue by MarcMagnin** [fhmq/hmq#17]
It would be nice to be able to use the broker as a dep as well.
I'll look into how to make it generic enough to satisfy such requirement.
@chowyu08 what do you think of it?

**Comment by MarcMagnin** [comment]
@chowyu08 I've done a change for this in [sync_pool](../../hmq/tree/sync_pool) if you can have a look.

**Comment by chowyu08** [comment]
yeah

**Comment by MarcMagnin** [comment]
@chowyu08 I'm not sure to understand, can you elaborate on "library can not let broker deploy independently"?
The goal of that issue was to be able to use the broker inside another project. It's seems to work fine.

**Comment by gallex** [comment]
Yes, I think it very important to do this .  I think it world has following feature : 
1. APIs to manage plugins .
2. manage log
